app/assets/javascripts/application.js.erb

```diff
@@ -11,3 +13,55 @@ window.addEventListener("DOMContentLoaded", () => {
     window['ga-disable-' + GA_ID] = true;
   }
 });
+
+const getPasswordStrength = password => {
+  const result = zxcvbn(password);
+  const score = 100 * result.score / 4;
+  const scoreText = score < 50 ? 'weak' : (score < 100 ? 'medium' : 'good');
+  const rawSuggestion = result.feedback.suggestions.join(', ');
+  // zxcvbn-js sometimes fails to return a suggestion even when the password is too weak, so we need to perform some coercion.
+  const suggestion = (!rawSuggestion) ? 'Choose a password that is harder to guess' : rawSuggestion;
+  return { score, scoreText, suggestion };
+};
+
+if ($('input[name="user[password]"]').length) {
+  $('body').on('keyup', 'input[name="user[password]"]', function(e) {
+    const { score, scoreText } = getPasswordStrength(this.value)
+    const feedback = `Password strength: ${scoreText}`;
+    if (this.value.length) {
+      $('.password-strength')
+        .css('display', 'inline-block')
+        .text(feedback);
+    } else {
+      $('.password-strength').css('display', 'none');
+    }
+  });
+
+  $('body').on('blur', 'input[name="user[password]"]', function(e) {
+    const { score, suggestion } = getPasswordStrength(this.value)
```

Comment:
sorry I'm late to the party, but this should really just send an AJAX request to the server, which has all the necessary artillery to compute password strength without having the `zxcvbn` on both sides of the wire.

```
on blur do
  send value to server
    get score and suggestion back
    offer feedback
```

Reply:
That's a good option, but wonder how the user experience would be due to the delay. May be worth giving it a try @alan-gds ?

Reply:
fixed now.